spec.md

```diff
@@ -0,0 +1,1679 @@
+
+## Terminology
+
+| Term              | Definition                                       |
+|-------------------|--------------------------------------------------|
+| Volume            | A unit of storage that will be made available inside of a CO-managed container, via the CSI.                        |
+| Block Volume      | A volume that will appear as a block device inside the container.                                                   |
+| Mounted Volume    | A volume that will be mounted using the specified file system and appear as a directory inside the container.       |
+| CO                | Container Orchestration system, communicates with Plugin Containers using CSI service RPCs.                         |
+| SP                | Storage Provider, the vendor of a CSI plugin implementation.                                                        |
+| RPC               | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).                                       |
+| Node              | A host where the user workload will be running, uniquely identifiable from the perspective of a Plugin by a NodeID. |
+| Plugin            | Aka “plugin implementation”, a gRPC endpoint that implements the CSI Services.                                      |
+| Plugin Supervisor | Process that governs the lifecycle of a Plugin, MAY be the CO.                                                      |
+
+## Objective
+
+To define an industry standard “Container Storage Interface” (CSI) that will enable storage vendors (SP) to develop a plugin once and have it work across a number of container orchestration (CO) systems.
+
+### Goals in MVP
+
+The Container Storage Interface (CSI) will
+
+* Enable SP authors to write one CSI compliant “plugin” that “just works” across all COs that implement CSI.
+* Define API (RPCs) that enable:
+  * Dynamic provisioning and deprovisioning of a volume.
+  * Attaching or detaching a volume from a node.
+  * Mounting/unmounting a volume from a node.
+  * Consumption of both block and mountable volumes.
+  * Local storage providers (e.g., device mapper, lvm).
+* Define plugin protocol RECOMMENDATIONS.
+  * How to share secrets/credentials with container.
+  * How to discover installed plugins.
+  * Container deployment requirements (CAP_SYS_ADMIN, mount namespace, container ENTRYPOINT, container ports to expose, etc.)
+
+### Non-Goals in MVP
+
+The Container Storage Interface (CSI) explicitly will not define, provide, or dictate in v1:
+
+* How a CO should manage plugin container lifecycle, including:
+  * How to maintain state (e.g. what is attached, mounted, etc.).
+  * How to deploy, install, upgrade, uninstall, monitor, or respawn (in case of unexpected termination) plugin containers.
+* A mechanism for a CO to define “grades of storage” (aka a “storage catalogue”).
+* Protocol-level authn/authz.
+* Packaging of the plugin.
```

Comment:
Yeah, I think the consensus is that we don't want to dictate Plugin packaging. Downgrade to SHOULD sounds good to me.